iotileship/iotile/ship/autobuild/ship_file.py

```diff
@@ -36,6 +36,19 @@ def autobuild_shiparchive(src_file):
     target = family.platform_independent_target()
     resolver = ProductResolver.Create()
 
+    #Parse through build_step products to see what needs to imported
+    build_steps = family.tile.build_steps()
+    custom_steps = []
+    for build_step in build_steps:
+        full_file_name = build_step.split(":")[0]
+        basename = os.path.splitext(os.path.basename(full_file_name))[0]
+        folder = os.path.dirname(full_file_name)
+
+        fileobj, pathname, description = imp.find_module(basename, [folder])
+        mod = imp.load_module(basename, fileobj, pathname, description)
+        custom_steps += filter(lambda c: c[1].__module__==basename, inspect.getmembers(mod, inspect.isclass))
```

Comment:
Two comments on this line:

1. It is better to use a list comprehension than a `filter` statement because it's more clear what is going on.  You could rewrite this line as:
```python
custom_steps += [x for x in inspect.getmembers(mod, inspect.isclass) if x[1].__module__ == basename)]
```

2. Unlike the proxy examples in HardwareManager, you know exactly what the build step class name is since it's the part of `build_step` after the colon.  So, you can just directly get it from the module without needing the `inspect` part:

```python
full_file_name, class_name = build_step.split(":")
custom_step = getattr(mod, class_name)
```

Reply:
oh wow, good point, fixed